intents/intent.ridl

```diff
@@ -151,15 +151,21 @@ struct IntentDataGetIdToken
 struct TransactionRaw
   - type: string
   - to: string
-  - value: string
+  - value?: string
   - data: string
 
+struct AbiData
```

Comment:
new AbiData struct for the "data" field where we expect an abi definition.

we could also call this, ContractCallData perhaps? maybe that is better name..?

also "func" field is optional, because if the "abi" field is just a function, its unnecessary.

Reply:
IMO `AbiData` is just fine 👌